lucene/core/src/java/org/apache/lucene/search/FilteringDocLeafComparator.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search;
+
+import org.apache.lucene.index.LeafReaderContext;
+
+import java.io.IOException;
+
+/**
+ * This comparator is used when there is sort by _doc asc together with "after" FieldDoc.
+ * The comparator provides an iterator that can quickly skip to the desired "after" document.
+ */
+public class FilteringDocLeafComparator implements FilteringLeafFieldComparator {
```

Comment:
Maybe rename to `AfterDocLeafComparator` ?

Reply:
I like `AfterDocLeafComparator`, but I renamed to `FilteringAfterDocLeafComparator` for consistency with all other filtering comparators. Please let me know if you still like it to be renamed